src/lib/variables/variables.ts

```diff
@@ -134,34 +133,42 @@ export const getValue = (
   assertIsVariables(result);
 
   return result;
-};
+}
 
 /**
  * Turns a variable (both primitive and non-primitive) into a string.
  */
-export const variableToString = (variable: Variables): string => {
+export function variableToString(variable: Variables): string {
   if (typeof variable === 'string') {
     return variable;
   }
 
+  if (variable === undefined) {
+    return 'undefined'; // TODO: for both undefined and null, return 'None'?
+  }
+
+  if (Buffer.isBuffer(variable)) {
+    return variable.toString();
+  }
+
   return JSON.stringify(variable);
-};
+}
 
 /**
  * Stringifies a Record of variables. `undefined` values are removed.
  */
-export const variablesToStrings = (
-  variables?: Variables
-): Record<string, string> => {
+export function variablesToStrings(
+  variables: Variables
```

Comment:
question: Looking at the documentation of this function now - should this even be taking `Variables` as input? Shouldn't this be taking `NonPrimitive` instead - if the input should always be a record...

If so, then the first if can be removed.

It could mean that we would need to update call sites.

Reply:
I started with it. Let's see if that's the right thingy